File: frontend/types/audit.ts
// Tipos para la bitácora de auditoría (CU-31). Espejo de
// backend/app/schemas/audit.py.

export interface AuditLog {
  id: string
  actor_user_id: string | null
  action: string
  entity_type: string
  entity_id: string | null
  payload_diff: Record<string, unknown> | null
  ip_address: string | null
  user_agent: string | null
  tenant_id: string
  created_at: string
}

export interface AuditLogListResponse {
  items: AuditLog[]
  total: number
  page: number
  limit: number
}

export interface AuditFilters {
  entity_type?: string
  entity_id?: string
  actor?: string
  action?: string
  from?: string
  to?: string
  page?: number
  limit?: number
}

File: backend/app/schemas/audit.py
"""Pydantic schemas for the audit log (CU-31)."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class AuditLogOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    actor_user_id: UUID | None
    action: str
    entity_type: str
    entity_id: UUID | None
    payload_diff: dict | None
    ip_address: str | None
    user_agent: str | None
    tenant_id: UUID
    created_at: datetime


class AuditLogListResponse(BaseModel):
    items: list[AuditLogOut]
    total: int
    page: int
    limit: int
